### PR TITLE
Move afl-fuzz related env variables into afl_state_t

### PR DESCRIPTION
**Pull request by rish9101**

### CODE REVIEW

File: src/afl-fuzz-init.c [hunk 1]
```diff
@@ -1486,7 +1486,7 @@ void check_crash_handling(void) {
       "    sudo launchctl unload -w ${SL}/LaunchDaemons/${PL}.Root.plist\n");
 
 #endif
-  if (!get_afl_env("AFL_I_DONT_CARE_ABOUT_MISSING_CRASHES"))
+  if (!)get_afl_env("AFL_I_DONT_CARE_ABOUT_MISSING_CRASHES")
```
Comment (domenukk) [inline src/afl-fuzz-init.c]: This looks ... Odd.

File: src/afl-fuzz-globals.c [hunk 2]
```diff
@@ -75,7 +76,7 @@ list_t afl_states = {0};
 
 /* Initializes an afl_state_t. */
 
-void afl_state_init(afl_state_t *afl) {
+void afl_state_init(afl_state_t *afl, char **envp) {
```
Comment (domenukk) [inline src/afl-fuzz-globals.c]: We can think about moving this out of the init but it's fine if we do it later, also

File: src/afl-fuzz-init.c (hunk 1 above)
Comment (rish9101) [reply src/afl-fuzz-init.c]: Ah. Didn't notice it. I'm sorry. I'll correct it. :sweat_smile:

File: src/afl-fuzz-globals.c (hunk 2 above)
Comment (rish9101) [reply src/afl-fuzz-globals.c]: I thought of making it a new function. I can do that if you say. :)

**Comment by rish9101** [comment]
Btw, if these changes are okay. I'll probably move the other environment variables (for afl-analyze, afl-tmin etc) out into appropriate structs too.

File: src/afl-fuzz-globals.c [hunk 3]
```diff
@@ -119,6 +120,221 @@ void afl_state_init(afl_state_t *afl) {
...
+
+void set_afl_environment(afl_state_t *afl, char **envp) {
+
+  int   index = 0, found = 0;
+  char *env;
+  while ((env = envp[index++]) != NULL) {
+
+    if (strncmp(env, "ALF_", 4) == 0) {
+
+      WARNF("Potentially mistyped AFL enqqqvironment variable: %s", env);
```
Comment (domenukk) [inline src/afl-fuzz-globals.c]: Can you change this typo? :)

File: src/afl-fuzz.c [hunk 4]
```diff
@@ -762,22 +762,22 @@ int main(int argc, char **argv_orig, char **envp) {
 
   }
 
-  if (get_afl_env("AFL_NO_FORKSRV")) afl->no_forkserver = 1;
-  if (get_afl_env("AFL_NO_CPU_RED")) afl->no_cpu_meter_red = 1;
-  if (get_afl_env("AFL_NO_ARITH")) afl->no_arith = 1;
-  if (get_afl_env("AFL_SHUFFLE_QUEUE")) afl->shuffle_queue = 1;
-  if (get_afl_env("AFL_FAST_CAL")) afl->fast_cal = 1;
+  if (afl->afl_env.afl_no_forksrv) afl->no_forkserver = 1;
```
Comment (domenukk) [inline src/afl-fuzz.c]: I think all of these are redundant.
Of they are only read here onc, we might not need a second .env variable 🤔 what's everybody's opinion on this?

File: src/afl-fuzz-globals.c (hunk 3 above)
Comment (rish9101) [reply src/afl-fuzz-globals.c]: Done. Sorry for this :sweat_smile:

File: src/afl-fuzz.c (hunk 4 above)
Comment (rish9101) [reply src/afl-fuzz.c]: That is true. Then, we'd have to add flags for it, right? Or, can we remove the variables from afl_state and keep them in env_struct?

Comment (domenukk) [reply src/afl-fuzz.c]: Well in these cases (`no_arith`, `shuffle_queue`, `fast_cal`, `no_forkserver`) there are already some flags.
I'd suggest we leave those out of the env struct for now, then decide what to do for the others later.

**Comment by domenukk** [comment]
@andreafioraldi Any idea, where `[!] afl-fuzz is not working correctly with qemu_mode compcov` comes from in Travis? I don't see any changes that could have affected this.

File: src/afl-fuzz.c [hunk 5]
```diff
@@ -681,8 +683,6 @@ int main(int argc, char **argv_orig, char **envp) {
         "Using -M master with the AFL_CUSTOM_MUTATOR_ONLY mutator options will "
         "result in no deterministic mutations being done!");
 
-  check_environment_vars(envp);
```
Comment (domenukk) [inline src/afl-fuzz.c]: What did this do and where did it go?

**Comment by domenukk** [comment]
@hexcoder- this would be an important step to get everything running in OpenBSD again, however I have no clue why Travis fails. Maybe you have an idea :)

**Comment by domenukk** [comment]
Ah no nevermind I mixed different things up 🙃 but still I have no clue

**Comment by vanhauser-thc** [comment]
> @andreafioraldi Any idea, where `[!] afl-fuzz is not working correctly with qemu_mode compcov` comes from in Travis? I don't see any changes that could have affected this.

My guess it was just unlucky that cmplog could not solve the test in time. I restarted that build check

**Comment by vanhauser-thc** [comment]
yes that fixed travis

**Comment by vanhauser-thc** [comment]
yes that fixed travis

**Pull request merged**